src/daft-core/src/series/ops/arithmetic.rs

```diff
@@ -99,17 +140,36 @@ impl Sub for &Series {
             InferDataType::from(self.data_type()).sub(InferDataType::from(rhs.data_type()))?;
         let lhs = self;
         match &output_type {
+            // ----------------
+            // Python
+            // ----------------
```

Comment:
these match statements were very difficult to comprehend. the comments IMO make it much easier to follow the matching logic.